### PR TITLE
Update CSS, make a rough same theme

### PR DESCRIPTION
**Pull request by flora8984461**
Merge-in #21 and make sure that we can run the full-stack

Part of https://github.com/EclipseFdn/react-eclipsefdn-members/issues/17

More issue:

- [x] I need to separate the states for new forms and existing forms

Please check the console log when each step finished (clicking on the next), there is error POST request with Fake endpoint each step.

More detailed CSS needs adjustments. 

- [x] Company Information
- [x]  Membership Level
- [x]  Working Group
- [x]  Signing Auth
- [x]  Review
- [x]  the Steppers
- [x]  Checkbox for Company Information - Same as Company Rep.
- [x]  What to do after selected Same as Company Rep.?
- [x]  Working Group - Add Another Working group
- [x]  Orange span *
- [x] More: more styles for the date picker.
- [x] Login Page
- [x] Add LESS file and wrap the whole membership form styles into a parental style, as Martin suggested
- [x] More: Validation on each step
- [x] After Submission Page

More:
- [x] Next step for myself to consider: think about making each page a sub-form so that I set initial values by POST and FETCH data easily?
- [x] Get rid of Router, open the app shows the first step, click on login, fetch user Id, and display all the forms he has, with the choice to start a new form. 

Questions coming:
I resolved it ~~1) Hide/disable when check "Same as Company Rep"? - to consider, before user inputs company rep, the user checks the box first. If using disable but showing the input box, do I need to sync when they type in company Rep if the user ticks the checkbox first? Or just leave it empty?~~

2) How to style for the "delete this working group" If more working groups added. (We need "delete" for each of the working groups entry to identify which to delete)

3) Defining what fields are required, what fields have more restrictions when validating the input.

4) Need the small orange icons for steppers as in the design

5) Country / Province selections

~~Don't mind about /login, probably broken now~~

Look at this link:
https://deploy-preview-21--eclipsefdn-react-members.netlify.app
~~https://deploy-preview-21--eclipsefdn-react-members.netlify.app/form~~

Signed-off-by: Yi Liu <yi.liu@eclipse-foundation.org>

### CODE REVIEW

**Comment by autumnfound** [comment]
We should encapsulate the CSS to make sure these selectors aren't destructive if ever included on a page with other structures. So any styles that affect a react form should be wrapped in something like `.eclipsefdn-react-form` to ensure only those elements are affected.

For reference, we implement our CSS somewhere between SMACSS and OOCSS  https://css-tricks.com/methods-organize-css/

File: src/components/forms/CustomStepButton.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const CustomStepButton = ({step, isSubmitting, setStep, isLastStep}) => {
           disabled={isSubmitting}
           // variant="contained"
           // color="primary"
-          className="btn btn-primary float-right"
+          className={`btn ${isLastStep() ? `btn-secondary` : `btn-primary `}`}
```
Comment (autumnfound) [inline src/components/forms/CustomStepButton.js:26]: rather than hardcode the button, a way we could do this is provide an override. If set, it would always have a given class, otherwise it would fallback to this. I'm not sure if that works with your current setup though, as I don't know if you have separate button calls, or if the submit/next button is just a single button outside the frame that is being redrawn.

File: src/components/forms/SignIn.js [hunk 2]
```diff
@@ -4,7 +4,10 @@ const SignIn = () => {
     
     return(
         <>
-            This is SignIn Page
+            <div className="text-center margin-bottom-20">
+                <a href="https://accounts.eclipse.org/" className="btn btn-secondary">Sign In</a>
```
Comment (autumnfound) [inline src/components/forms/SignIn.js:8]: Do we want buttons like this, or should we be calling the button components we have? I'm guessing that we have one, not sure if we do or not

File: src/components/forms/SignInJumbotron.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
```
Comment (autumnfound) [inline src/components/forms/SignInJumbotron.js:1]: If we are using the Hugo theme, all we need to do is set stuff like headline to get the jumbotron. We shouldn't be implementing things we already have

Comment (flora8984461) [reply src/components/forms/SignInJumbotron.js:1]: Oh, this one is not the real jumbotron, I'm calling it jumbotron since I don't know what name to call 😅

It's this part 
image

Comment (autumnfound) [reply src/components/forms/SignInJumbotron.js:1]: We could use the term `introduction`, `preamble`, or break it down into a more basic concept like `image-block` to denote that its an image + text block

Comment (flora8984461) [reply src/components/forms/SignInJumbotron.js]: I changed the name

**Comment by flora8984461** [comment]
> We should encapsulate the CSS to make sure these selectors aren't destructive if ever included on a page with other structures. So any styles that affect a react form should be wrapped in something like `.eclipsefdn-react-form` to ensure only those elements are affected.
> 
> For reference, we implement our CSS somewhere between SMACSS and OOCSS https://css-tricks.com/methods-organize-css/

to take a note, this is done, I wrapped them

File: src/components/forms/SignIn.js (hunk 2 above)
Comment (flora8984461) [reply src/components/forms/SignIn.js]: I think that one is type="button" for my form steps, not for go to the link. 🤔

File: src/components/forms/CustomStepButton.js (hunk 1 above)
Comment (flora8984461) [reply src/components/forms/CustomStepButton.js:23]: Is not this what currently doing? when passing `isLastStep` is to override or use the default class. 🤔

Comment (autumnfound) [reply src/components/forms/CustomStepButton.js:23]: Its more that this is less usable outside the context of this exact form on this site. What if we need different configs for different site forms? What if button style means we need different classes on the button on a different site? There is no way to inject a different button class to be used in these instances. This is fine for this form, but its good to keep in mind when writing reusable code that we usually like the ability to fine tune stuff like this. Maybe make an issue to track this as an enhancement

**Comment by autumnfound** [comment]
Overall feedback, CSS class fake-input should probably be named preview-field or read-only-input. It's more accurate overall for the use-cases.

Additionally, we should remove any commented out code and debug console logs. It makes our forms very noisy and isn't typically needed. We should console log when something goes wrong, rather than as testing in committed code. There's a decent amount of console logs, and we should probably strip most of it out.

**Comment by flora8984461** [comment]
> Overall feedback, CSS class fake-input should probably be named preview-field or read-only-input. It's more accurate overall for the use-cases.
> 
> Additionally, we should remove any commented out code and debug console logs. It makes our forms very noisy and isn't typically needed. We should console log when something goes wrong, rather than as testing in committed code. There's a decent amount of console logs, and we should probably strip most of it out.

I have removed those for debugging purposes, kept some mock data POST/PUT -> it's when you click on "Next" button. And kept some information comments.

**Comment by flora8984461** [comment]
> > Overall feedback, CSS class fake-input should probably be named preview-field or read-only-input. It's more accurate overall for the use-cases.
> > Additionally, we should remove any commented out code and debug console logs. It makes our forms very noisy and isn't typically needed. We should console log when something goes wrong, rather than as testing in committed code. There's a decent amount of console logs, and we should probably strip most of it out.
> 
> I have updated the CSS name, removed those for debugging purposes, kept some mock data POST/PUT -> it's when you click on "Next" button. And kept some information comments.

Wait a min, have some more cleanup...

**Comment by flora8984461** [comment]
Now should be ready. I took this https://github.com/EclipseFdn/react-eclipsefdn-members/pull/21#discussion_r577073689 to be a separate issue for enhancement

File: src/Constants/Constants.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// list all constants here
+export const FETCH_METHOD = {
```
Comment (autumnfound) [inline src/Constants/Constants.js:2]: These should be single quoted comments, not double quoted (consistency, would fail linting)

File: src/components/FormComponents/WorkingGroups/ParticipationLevel.js [hunk 5]
```diff
@@ -8,12 +9,7 @@ const ParticipationLevel = ({name, workingGroup}) => {
   useEffect(() => {
     let isSubscribed = true;
 
-    fetch("workingGroups.json", {
-        headers : { 
-          'Content-Type': 'application/json',
-          'Accept': 'application/json'
-        }
-      })
+    fetch("workingGroups.json", { headers : FETCH_HEADER })
```
Comment (autumnfound) [inline src/components/FormComponents/WorkingGroups/ParticipationLevel.js:12]: should be single quoted (though this one doesn't matter much, as the value will change eventually)

File: src/components/FormComponents/formModels/formFieldModel.js [hunk 6]
```diff
@@ -1,3 +1,17 @@
+const firstName = "First Name"
```
Comment (autumnfound) [inline src/components/FormComponents/formModels/formFieldModel.js:1]: Strings should be single quoted not double quoted

**Review by autumnfound — COMMENTED** [review]
Other than linting issues, this looks great!

File: src/components/FormComponents/WorkingGroups/ParticipationLevel.js (hunk 5 above)
Comment (flora8984461) [reply src/components/FormComponents/WorkingGroups/ParticipationLevel.js:12]: So for passing props to component, such as:

```
      <Select
        label="ParticipationLevel"
        name={name}
        options={participationLevels}
      />
```

Do I need to change all of them to use single quote? Or just when I define strings?

Comment (flora8984461) [reply src/components/FormComponents/WorkingGroups/ParticipationLevel.js:12]: And also about import:
```
import Select from "../Inputs/Select";
import { FETCH_HEADER } from '../../../Constants/Constants';
```

File: src/components/FormComponents/formModels/formFieldModel.js (hunk 6 above)
Comment (flora8984461) [reply src/components/FormComponents/formModels/formFieldModel.js:1]: Actually, Is there any standard to follow using single quotes?

Comment (autumnfound) [reply src/components/FormComponents/formModels/formFieldModel.js]: It's a consistency thing and applies to Javascript, not HTML. https://bytearcher.com/articles/single-or-double-quotes-strings-javascript/ is a quick explanation on it. 

TLDR, most people use single quotes. No difference, but consistency is important. Our linting (when active) has an error condition on double-quotes. Keep in mind JSON files and literal JSON strings need double quotes, but there are no cases of that in the files I have pointed out thankfully!

File: src/components/FormComponents/WorkingGroups/ParticipationLevel.js (hunk 5 above)
Comment (autumnfound) [reply src/components/FormComponents/WorkingGroups/ParticipationLevel.js]: Components are fancy JSX, and it looks like the convention is double-quoting. This is due to the convention that JSX is meant to be as close to HTML as possible (which prefers double-quotes).

File: src/components/FormComponents/formModels/formFieldModel.js (hunk 6 above)
Comment (flora8984461) [reply src/components/FormComponents/formModels/formFieldModel.js]: To keep consistance, 

When importing other components, use Double Quote;
When passing props to Components, such as `<input label="" foo="">`, Use  Double Quote ?
When call functions and passing strings, and when defining string variables, Use single Quote?

Comment (autumnfound) [reply src/components/FormComponents/formModels/formFieldModel.js]: Anytime you are doing things that are standard JS operations, including imports, use single quotes. When doing react component JSX (the things like `<input labelName="" foo="" />`) feel free to use double quotes

Comment (flora8984461) [reply src/components/FormComponents/formModels/formFieldModel.js]: Updated.

**Comment by flora8984461** [comment]
Will be included in #47 .

**Comment by autumnfound** [comment]
Feedback on PR, not all is actionable, but just to track points of discussion to branch into issues or fixes in this PR if applicable.

Functionality:
- Clicking the steps at the top should update form state in the same way that the next button does. Some of the contact syncing (i.e. using the same as company rep. check) doesn't work as intended when using the form steps at the top to navigate.
- In the review, we should display the working group to join by the display name, not the ID (as it may confuse end users).
- WG rep should also be hidden to match requirements from marketing (unless this shifted outside of the XD doc).

Accessibility:
- Fields are not connected to labels (looks like you used name instead of ID, which is [how labels connect](https://www.w3schools.com/tags/tag_label.asp))
- Orange required stars are low visibility, we may want to make them bigger and bold (design, not implementation issue)

Additional comments:
- We should be able to navigate the form backwards and forwards at will (with a loading caption if necessary), currently form traversal is limited to form steps that have been already opened.
- We should probably create a gap or panel to contain each wg that is being joined, and modify label to be `Working Group(s) to join`
image

**Comment by flora8984461** [comment]
`We should be able to navigate the form backwards and forwards at will (with a loading caption if necessary), currently form traversal is limited to form steps that have been already opened.` For this one, I think this is expected behaviour, you don't want the user to go to steps that before previous steps are finished or have bad inputs. 🤔

An example could be https://enrollment.lfx.linuxfoundation.org/?project=tlf#/step-1

**Comment by chrisguindon** [comment]
I agree with most of your feedback @autumnfound .

The only exception is:

We should be able to navigate the form backwards and forwards at will (with a loading caption if necessary), currently form traversal is limited to form steps that have been already opened.

Users should not be allowed to move forward if required data is missing. The moving background should be allowed.

YI, please move forward with those changes.

**Comment by autumnfound** [comment]
> I agree with most of your feedback @autumnfound .
> 
> The only exception is:
> 
> > We should be able to navigate the form backwards and forwards at will (with a loading caption if necessary), currently form traversal is limited to form steps that have been already opened.
> 
> Users should not be allowed to move forward if required data is missing. Moving backward should be allowed.
> 
> YI, please move forward with those changes.

Apologies, context was I loaded a completed form in a fresh session

**Comment by flora8984461** [comment]
`Actually, I still need to make user can navigate between finished if they continue an existing form (and it has several steps finished)`

I think this is not that easily added, I think this is an enhancement later.

**Comment by flora8984461** [comment]
@autumnfound  For `Fields are not connected to labels (looks like you used name instead of ID, which is how labels connect)`, do I just add the id to <input> will do? I am passing the Fieldname (this is formik feature, it recognize which field to update by field name). I am thinking to add id = {fieldname} as well.

**Comment by flora8984461** [comment]
@autumnfound Some of the inputs, I am not using label, while using `h2 / h3`, such as in Working Group Date and Working Group select. In that case, should I change those h2/h3 to use `<label>`? Or add a hidden label with `sr-only` ?

**Comment by autumnfound** [comment]
> @autumnfound Some of the inputs, I am not using label, while using `h2 / h3`, such as in Organization Select and Working Group select. In that case, should I change those h2/h3 to use `<label>`? Or add a hidden label with `sr-only` ?

see aria attributes https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

**Comment by flora8984461** [comment]
> > @autumnfound Some of the inputs, I am not using label, while using `h2 / h3`, such as in Organization Select and Working Group select. In that case, should I change those h2/h3 to use `<label>`? Or add a hidden label with `sr-only` ?
> 
> see aria attributes https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

So it's to add `id` to h2/h3, and mark my input / select `aria-labelledby` with that id

**Comment by flora8984461** [comment]
@autumnfound Updated all the feedbacks, except for https://github.com/EclipseFdn/react-eclipsefdn-members/issues/48 

Thanks for the feedback!

**Review by autumnfound — APPROVED** [review]
LGTM :) The issues I mentioned have been fixed
